client.go

```diff
@@ -0,0 +1,74 @@
+package filetypes
+
+import (
+	"github.com/cloudquery/filetypes/csv"
+	"github.com/cloudquery/filetypes/json"
+)
+
+type FormatType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type FileSpec struct {
+	Format         FormatType `json:"format,omitempty"`
+	IncludeHeaders bool       `json:"include_headers,omitempty"`
+	Delimiter      rune       `json:"delimiter,omitempty"`
+}
```

Comment:
I think we should consider having a Spec struct per file type, rather than having a single struct contain all the options for all current and future file types. This way it is clear which options apply to which formats, and would be more consistent with the rest of our config style. From a user point of view the spec might look like this:

```
kind: destination
spec:
  name: "file"
  path: "cloudquery/file"
  version: "v1.0.5"
  write_mode: "append" # file only supports 'append' mode
  spec:
    directory: "./cq_csv_output"
    format: "csv"
    format_spec:
         include_headers: true
         delimeter: ","
```

In our code, something like this:

```
type CSVSpec struct {
    IncludeHeaders bool `json:"include_headers,omitempty"`
    Delimiter rune `json:"delimiter,omitempty"`
}

type JSONSpec struct {
}
```

The `format` value would become the responsibility of the destination plugin spec. 

There are still a few potential file types and options to be added (Parquet, Avro, Excel CSV being top of mind for me) so I believe this separation will be worthwhile in the long run.